components/ui-api-layer/internal/domain/k8s/pod_converter.go

```diff
@@ -0,0 +1,187 @@
+package k8s
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/kyma-project/kyma/components/ui-api-layer/internal/gqlschema"
+	v1 "k8s.io/api/core/v1"
+)
+
+type podConverter struct{}
+
+func (c *podConverter) ToGQL(in *v1.Pod) (*gqlschema.Pod, error) {
+	if in == nil {
+		return nil, nil
+	}
+
+	containerStates := c.containerStatusesToGQLContainerStates(in.Status.ContainerStatuses)
+
+	gqlJSON, err := c.podToGQLJSON(in)
+	if err != nil {
+		return nil, err
+	}
+
+	return &gqlschema.Pod{
+		Name:              in.Name,
+		NodeName:          in.Spec.NodeName,
+		Namespace:         in.Namespace,
+		RestartCount:      c.getRestartCount(in.Status.ContainerStatuses),
+		CreationTimestamp: in.CreationTimestamp.Time,
+		Labels:            in.Labels,
+		Status:            c.podStatusPhaseToGQLStatusType(in.Status.Phase),
+		ContainerStates:   containerStates,
+		JSON:              gqlJSON,
+	}, nil
+}
+
+func (c *podConverter) ToGQLs(in []*v1.Pod) ([]gqlschema.Pod, error) {
+	var result []gqlschema.Pod
+	for _, u := range in {
+		converted, err := c.ToGQL(u)
+		if err != nil {
+			return nil, err
+		}
+
+		if converted != nil {
+			result = append(result, *converted)
+		}
+	}
+	return result, nil
+}
+
+func (c *podConverter) containerStatusesToGQLContainerStates(in []v1.ContainerStatus) []gqlschema.ContainerState {
+	containerStates := []gqlschema.ContainerState{}
+
+	for _, containerStatus := range in {
+		if containerStatus.State.Waiting != nil {
+			containerStates = append(containerStates, c.getWaitingContainerState(containerStatus.State.Waiting))
+		} else if containerStatus.State.Terminated != nil {
+			containerStates = append(containerStates, c.getTerminatedContainerState(containerStatus.State.Terminated))
+		} else if containerStatus.State.Running != nil {
+			containerStates = append(containerStates, c.getRunningContainerState())
+		} else {
+			containerStates = append(containerStates, c.getDefaultContainerState())
+		}
+	}
+
+	return containerStates
+}
+
+func (c *podConverter) getWaitingContainerState(in *v1.ContainerStateWaiting) gqlschema.ContainerState {
```

Comment:
Export logic responsible for extracting status to separate package, please see how it is created in `servicecatalog` domain.